homeassistant/components/lamarzocco/config_flow.py

```diff
@@ -0,0 +1,242 @@
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except NoMachines:
+                errors["base"] = "no_machines"
+
+            if not errors:
+                self._config = data
+                return await self.async_step_machine_selection()
+
+        return self.async_show_form(
+            step_id="user", data_schema=LOGIN_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_machine_selection(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Let user select machine to connect to."""
+        errors = {}
+        if user_input is not None:
+            serial_number = user_input[CONF_MACHINE]
+            await self.async_set_unique_id(serial_number)
+            self._abort_if_unique_id_configured()
+
+            # if host is set, check if we can connect to it
+            if user_input.get(CONF_HOST):
+                lm = LaMarzoccoClient(self.hass, self._config)
+                if not await lm.check_local_connection(
+                    credentials=self._config,
+                    host=user_input[CONF_HOST],
+                    serial=serial_number,
+                ):
+                    errors[CONF_HOST] = "cannot_connect"
+            if not errors:
+                return self.async_create_entry(
+                    title=serial_number, data=self._config | user_input
+                )
+
+        machine_options = [
+            SelectOptionDict(
+                value=serial_number,
+                label=f"{model_name} ({serial_number})",
+            )
+            for serial_number, model_name in self._machines
+        ]
+        machine_selection_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_MACHINE,
+                    default=machine_options[0],
+                ): SelectSelector(
+                    SelectSelectorConfig(
+                        options=machine_options,
+                        mode=SelectSelectorMode.DROPDOWN,
+                    )
+                ),
+                vol.Optional(CONF_HOST): cv.string,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="machine_selection",
+            data_schema=machine_selection_schema,
+            errors=errors,
+        )
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> FlowResult:
+        """Handle a flow initialized by discovery over Bluetooth."""
+        address = discovery_info.address
+        name = discovery_info.name
+
+        _LOGGER.debug(
+            "Discovered La Marzocco machine %s through Bluetooth at address %s",
+            name,
+            address,
+        )
+
+        self._discovered[CONF_NAME] = name
+        self._discovered[CONF_MAC] = address
+
+        await self.async_set_unique_id(address)
```

Comment:
Here, you set the unique_id to address, which will be overwritten in `async_step_machine_selection` with the serial number. This makes no sense.

Please explain when the Bluetooth discovery will be used?

Reply:
The idea was, that if I discover a device through Bluetooth, I can set the MAC in a config entry and don't need to scan for it on every restart. Most users will not know their Bluetooth MAC so that's why I'm scanning and only allow the user to set it in the options flow. 
As Bluetooth requires authentication, and we also need the cloud access, the user then needs to access their credentials.

Reply:
always using serial as unique_id now